### DISCUSSION TITLE
default 'as' in composite component.

### DISCUSSION THREAD

**Discussion by VaibhavAcharya** [stitchesjs/stitches#444] [Help] [answered]
Example:
```
const Text = styled('span', { ... });

export const Paragraph = styled(Text, { ... });
export const Heading = styled(Text, { ... });
```

Here, I would want the `Paragraph `element to render as `p` & `Heading` as `h1` ***by default***.

Is there a way to set as prop as default on composite component ?

**Comment by makezi** [comment, marked as answer]
`defaultProps` should work here:

```javascript
export const Paragraph = styled(Text, { ... });

Paragraph.defaultProps = {
  as: 'p'
}

```

**Reply by peduarte** [reply]
This should work but note React will deprecate it soon. 

Another option is to wrap it in your own component and pre-define the as prop.

**Comment by rafgraph** [comment]
@VaibhavAcharya check out #609

**Comment by bel0v** [comment]
This does not work with typescript though.
getting  `Type '{ as: string; }' is not assignable to type ...` error

[codesandbox demo](https://codesandbox.io/s/happy-dubinsky-suwphg?file=/src/App.tsx)
